gamemode/sv_player.lua

```diff
@@ -203,81 +203,243 @@ function GM:PlayerDeathSound()
-			ChosenSpawnPoint != self.LastSpawnPoint ) then
-			
-			if ( hook.Call( "IsSpawnpointSuitable", GAMEMODE, pl, ChosenSpawnPoint, i == Count ) ) then
-			
-				self.LastSpawnPoint = ChosenSpawnPoint
-				pl:SetVar( "LastSpawnpoint", ChosenSpawnPoint )
-				return ChosenSpawnPoint
-			
+
+	return new_table
+end
+
+
+-------------------------------
+-------------------------------
+-------------------------------
+
+-- Most of the following code from this comment block down to the next is from TTT.
+-- The code provides the functionality for automatically generating spawnpoints if
+-- necessary. Prophunters sort-of-not-really had a system for doing this but it
+-- basically just killed players every time or got them stuck in each other so it
+-- was not very useful. The TTT code has been tweaked to work better with Prophunters.
+
+
+-- Nice Fisher-Yates implementation, from Wikipedia
+local rand = math.random
+local function ShuffleTable(t)
+	local n = #t
+
+	while n > 2 do
+		-- n is now the last pertinent index
+		local k = rand(n) -- 1 <= k <= n
+		-- Quick swap
+		t[n], t[k] = t[k], t[n]
+		n = n - 1
+	end
+
+	return t
+end
+
+
+function GM:IsSpawnpointSuitable(ply, spwn, force, rigged)
+	if !IsValid(ply) || ply:Team() == 1 then return true end
+	if !rigged && (!IsValid(spwn) || !spwn:IsInWorld()) then return false end
+
+	-- spwn is normally an ent, but we sometimes use a vector for jury rigged
+	-- positions
+	local pos = rigged && spwn || spwn:GetPos()
+
+	if not util.IsInWorld(pos) then return false end
+
+	local blocking = ents.FindInBox(pos + Vector( -32, -32, 0 ), pos + Vector( 32, 32, 64 )) -- Changed from (-16, -16, 0) (16, 16, 64)
+
+	for _, blocking_ent in ipairs(blocking) do
+		if IsValid(blocking_ent) && blocking_ent:IsPlayer() && blocking_ent:Team() != 1 && blocking_ent:Alive() then
+			if force then
+				blocking_ent:Kill()
+			else
+				return false
 			end
-			
 		end
-			
 	end
-	
-	return ChosenSpawnPoint
-	
+
+	return true
+end
+
+
+-- TTT only had a single table for spawnpoints but we're going to use three different ones
+-- so that we can try to group teams together.
+local prop_spawn_types = {"info_player_terrorist", "info_player_axis",
+"info_player_combine", "info_player_pirate", "info_player_viking",
+"diprip_start_team_blue", "info_player_blue", "info_player_human"}
+
+local hunter_spawn_types = {"info_player_counterterrorist",
+"info_player_allies", "info_player_rebel", "info_player_knight",
+"diprip_start_team_red", "info_player_red", "info_player_zombie"}
+
+-- These spawn types should ideally only be used for spectators. Hunters/props spawning here
+-- will probably fall out of the world.
+local spectator_spawn_types = {"info_player_start", "gmod_player_start",
+"info_player_teamspawn", "ins_spawnpoint", "aoc_spawnpoint",
+"dys_spawn_point", "info_player_coop", "info_player_deathmatch"}
+
+
+function GetSpawnEnts(ply_team, shuffled, force_all)
+	local tbl_to_use
```

Comment:
This must all be from TTT, since it's using `snake_case`.

Reply:
All of the snake_case was totally my fault, I completely forgot that it should've been camelCase.